ratelimits/transaction.go

```diff
@@ -183,15 +183,27 @@ type TransactionBuilder struct {
 	*limitRegistry
 }
 
+// NewTransactionBuilderFromFiles returns a new *TransactionBuilder. The
+// provided defaults and overrides paths are expected to be paths to YAML files
+// that contain the default and override limits, respectively. Overrides is
+// optional, defaults is required.
+func NewTransactionBuilderFromFiles(defaults, overrides string) (*TransactionBuilder, error) {
+	registry, err := newLimitRegistryFromFiles(defaults, overrides)
+	if err != nil {
+		return nil, err
+	}
+	return &TransactionBuilder{registry}, nil
+}
+
 // NewTransactionBuilder returns a new *TransactionBuilder. The provided
-// defaults and overrides paths are expected to be paths to YAML files that
-// contain the default and override limits, respectively. Overrides is optional,
-// defaults is required.
-func NewTransactionBuilder(defaults, overrides string) (*TransactionBuilder, error) {
-	registry, err := newLimitRegistry(defaults, overrides)
+// defaults map is expected to contain default limit data. Overrides are not
+// supported. Defaults is required.
+func NewTransactionBuilder(defaults LimitConfigs) (*TransactionBuilder, error) {
+	registry, err := newLimitRegistry(defaults, nil)
 	if err != nil {
 		return nil, err
 	}
+
```

Comment:
consistency nit: Match local style for (no) white-space before the final return.

```suggestion
```